### DISCUSSION TITLE
auto import from typing instead of main source

### DISCUSSION THREAD

**Discussion by beauxq** [microsoft/pylance-release#2629] [Enhancement]
## Environment data

-   Language Server version: v2022.1.3
-   OS and version: Windows 10
-   Python version (& distribution if applicable, e.g. Anaconda): 3.10

## to reproduce:
open a new python file in vs code and type this: (type the last line, not copy and paste)
```python3
from typing import Literal


def f() -> None:
    a: Literal['abc'] = 'abc'
    b = Counte
```
Then press enter.

## Actual behaviour

After typing the last `e`, the autocomplete menu is visible with `Counter` selected.
Pressing enter changes the top line to:
```python3
from typing import Counter, Literal
```

## Expected behaviour

Pressing enter should add a new line at the top:
```python3
from collections import Counter
```

**Comment by beauxq** [comment]
If I type
```python3
b: Counte
```
THEN it should import from typing (only in older python versions < 3.9).
but
```python3
b = Counte
```
should always import from collections.

**Comment by heejaechang** [comment]
I am not sure the reasoning behind of what you are suggesting. both Counter can be used as type annotation and object creation (https://stackoverflow.com/questions/61551094/how-to-deal-with-name-clash-collections-counter-and-typing-counter#:~:text=Starting%20with%20Python%203.9%20you%20can%20do%3A%20from,%5B%5D.%20See%20PEP%20585%20and%20Generic%20Alias%20Type.)

that being said, auto import should show both Counter and it is up to you which one you decide to use.

actually, I can see the reasoning. so probably Enhancement?

**Comment by heejaechang** [comment]
I have put it in the backlog,

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.
